compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2221,10 +2221,11 @@ public BLangNode transform(IndexedExpressionNode indexedExpressionNode) {
             indexBasedAccess.expr = ((BLangGroupExpr) expression).expression;
             BLangGroupExpr group = (BLangGroupExpr) TreeBuilder.createGroupExpressionNode();
             group.expression = indexBasedAccess;
-            group.pos = getPosition(indexedExpressionNode);
+            group.pos = indexBasedAccess.pos;
```

Comment:
@sanjana Do we need this change to resolve this issue?

Reply:
@mohanvive the `indexBasedAccess.pos` is declared at https://github.com/ballerina-platform/ballerina-lang/pull/32941/files#diff-f19078e211ec6eaa361839b611670e633460cbe1f9193a55e6e11ffbce0409ecR2193 

I reused its value here in order to avoid re-computing `getPosition(indexedExpressionNode)`. 

It is not directly required for this issue. But, an optimization that I came across.

Reply:
Isn't this the wrong position to be setting for this? 🤔 IINM, the above is a case like the following: `<1>(expr)<2>[1]<3>`. For the group expr, the position should be `<1>` to `<2>`. The above is setting `<1>` to `<3>` for it

Reply:
Consider the example code `int a = <int> (dataMap)["a"];`, where the `group` is `(dataMap)["a"]`, and `group.expression`'s position is set for `(dataMap)`. 
Therefore, the above position setting can be considered as accurate. 
In that case, the position of `group` shall be `<1>` to `<3>`.

@lochana-chathura WDYT?

Reply:
As per L2223 `group` seem to present the whole indexed access expr here. so should the position?

Here the AST representation is confusing to me. Why would we create a `BLangGroupExpr` node here, when the actually node is not in the form of `(` `expression` `)`.  Ideally we need to create `BLangIndexBasedAccess` node with `bLangIndexBasedAccess.expr` being a `BLangGroupExpr`.

Reply:
Exactly. Basically that whole if block is unnecessary right?

Reply:
yeah, unnecessary.